src/main/java/ui/components/pickers/LabelPickerState.java

```diff
@@ -0,0 +1,267 @@
+package ui.components.pickers;
+
+import backend.resource.TurboLabel;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This class is used to represent the state of the label picker. In addition to representing a state,
+ * it contains the logic which handles toggling/highlighting of labels and simplifies the retrieval of
```

Comment:
Does it really handles `highlighting`? Should it? Highlighting is a presentation issue.
More accurate to say `it contains the state transition logic`?